test/broadcast.jl

```diff
@@ -164,8 +164,6 @@ m = [1:2;]'
 @test @inferred([0,1.2].+reshape([0,-2],1,1,2)) == reshape([0 -2; 1.2 -0.8],2,1,2)
 rt = Base.return_types(.+, Tuple{Array{Float64, 3}, Array{Int, 1}})
 @test length(rt) == 1 && rt[1] == Array{Float64, 3}
-rt = Base.return_types(broadcast, Tuple{typeof(.+), Array{Float64, 3}, Array{Int, 3}})
-@test length(rt) == 1 && rt[1] == Array{Float64, 3}
```

Comment:
With the changes broadcast wouldn't be inferable (same as map) but it will return the correct type nevertheless. I guess I should change it to at least verify that.